WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt

```
@@ -249,6 +252,37 @@ class ProductDetailViewModel @AssistedInject constructor(
         updateProductDraft(saleEndDate = Optional(null))
     }
 
+    fun onProductDownloadClicked(file: ProductFile) {
+        triggerEvent(
+            EditProductDownload(
+                file,
+                viewState.productDraft!!.downloadLimit,
+                viewState.productDraft!!.downloadExpiry
```

np: can we refrain from using `!!` here? The `productDraft` should **not** be null but **if** it is, it would be great if the app doesn't crash. We can pass a default `downloadLimit` and `downloadExpiry`, if those scenarios, which would be a better user experience.

---

Can we discuss about this 🙂? IMHO if we are sure that the `productDraft` should not be null (and when clicking on a downloadable file in the list, this is the case), and that if it's null, then something is wrong, we shouldn't let it fail silently by giving a default value, at least if it crashed we may gain an insight that our assumption that it shouldn't be null is not correct, or not working always, and we need to do something about it.

---

I agree the `productDraft` model should ideally never be null and if it is, we should have a mechanism to let us know that something is not working :) But IMHO, crashing the app to let us know something is not working is never a good user experience. 

But the worst case scenario that can happen if we have a default value here is that the user would need to refresh the product screen again to view the correct value. We could even add a warning here to let them know that we are unable to display the correct value here. It would be better than the app crashing.

And to debug what is wrong, we could add a custom field to Sentry to let us know that the `productDraft` is null, when it shouldn't be. We would be able to have all the details to debug and it would not be disruptive to the user as well. WDYT?

We could add a custom event in Sentry that could let us know if the `productDraft` is null, when it shouldn't be, in all of the product editing screens.

---

These changes are no longer necessary because of the updates suggested by Garance 🙂
I agree with that a crash is never a good user experience, but deciding between what's best for the user, is it a wrong state of the app or a crash, I would say it depends on the app, and here dealing with changes that may be pushed, I'd say a crash is better than overriding the user's values 🙂, but neither one of those would happen, because the value won't be null.
And it's not something that we don't currently use in the app: https://github.com/woocommerce/woocommerce-android/blob/develop/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/categories/ProductCategoriesFragment.kt#L138

> We could even add a warning here to let them know that we are unable to display the correct value here. It would be better than the app crashing.

I fully agree with this solution as the better one 👍